test/rpc/validation.js

```diff
@@ -0,0 +1,59 @@
+const test = require('tape')
+const request = require('supertest')
+
+const { startRPC, closeRPC } = require('./helpers')
+const { middleware } = require('../../lib/rpc/validation')
+
+test('should comlete the request with non-params if the method does not require any params', t => {
+  const server = startRPC({
+    'mock': middleware((params, cb) => {
+      cb()
+    }, 0, [])
+  })
+  const req = {
+    jsonrpc: '2.0',
+    method: 'mock',
+    id: 1
+  }
+
+  request(server)
+    .post('/')
+    .set('Content-Type', 'application/json')
+    .send(req)
+    .expect(res => {
+      if (res.body.error) {
+        throw new Error('should not return an error object')
+      }
+    })
+    .end(err => {
+      closeRPC(server)
+      t.end(err)
+    })
+})
+
+test('should throw if the method requires params but non-params', t => {
```

Comment:
Here a bit the same, suggestion:

'...if the API endpoint requires parameters as an input but no "params" request parameter is provided'